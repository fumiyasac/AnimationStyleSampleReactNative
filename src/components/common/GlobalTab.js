/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import {
  StyleSheet
} from 'react-native';

import { 
  Button,
  Icon,
  Text
} from 'native-base';

export default class GlobalTab extends React.Component {
  render() {
    return (
      <Button vertical onPress={this.props.onPress}>
        <Icon style={this.props.selected ? styles.tabSelectedColor : styles.tabDefaultColor} name={this.props.icon} />
        <Text style={this.props.selected ? styles.tabSelectedColor : styles.tabDefaultColor}>{this.props.title}</Text>
      </Button>
    );
  }
}

const styles = StyleSheet.create({
  tabDefaultColor: {
    color: '#999999',
  },
  tabSelectedColor: {
    color: '#4169e1'
  }
});