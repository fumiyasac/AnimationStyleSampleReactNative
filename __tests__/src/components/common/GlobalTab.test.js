import 'react-native';
import React from 'react';
import GlobalTab from '../../../../src/components/common/GlobalTab';

import renderer from 'react-test-renderer';

it('GlobalTabコンポーネントが正しく表示されること', () => {

  // 表示対象のオブジェクト
  let stubObject = { screen: "feed", title: "フィード", icon: "list" };

  // onPressのモック
  const onPressMock = jest.fn();

  // スナップショットの書き出し
  const tree = renderer.create(
    <GlobalTab 
      key={0}
      selected={true}
      title={stubObject.title}
      icon={stubObject.icon}
      onPress={onPressMock()} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
  