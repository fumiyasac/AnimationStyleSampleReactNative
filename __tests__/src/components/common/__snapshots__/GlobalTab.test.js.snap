// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalTabコンポーネントが正しく表示されること 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#007aff",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "borderWidth": undefined,
      "elevation": 2,
      "flexDirection": "column",
      "height": null,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": undefined,
      "shadowOffset": undefined,
      "shadowOpacity": undefined,
      "shadowRadius": undefined,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#fff",
            "fontSize": 24,
            "marginLeft": 16,
            "marginRight": 16,
            "paddingTop": 2,
          },
          Object {
            "color": "#4169e1",
          },
        ],
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        Object {
          "color": "#4169e1",
        },
      ]
    }
    uppercase={false}
  >
    フィード
  </Text>
</View>
`;
